src/Infrastructure.VS/VsVersion/VsVersionProvider.cs

```diff
@@ -32,24 +31,26 @@ namespace SonarLint.VisualStudio.Infrastructure.VS.VsVersion
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class VsVersionProvider : IVsVersionProvider
     {
-        public IVsVersion Version { get; }
+        private readonly Lazy<IVsVersion> lazyVersion;
+
+        public IVsVersion Version => lazyVersion.Value;
 
         [ImportingConstructor]
-        public VsVersionProvider([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, ILogger logger)
-            : this(serviceProvider, new SetupConfigurationProvider(), logger)
+        public VsVersionProvider(IVsUIServiceOperation serviceOperation, ILogger logger)
+            : this(serviceOperation, new SetupConfigurationProvider(), logger)
         {
         }
 
-        internal VsVersionProvider(IServiceProvider serviceProvider, ISetupConfigurationProvider setupConfigurationProvider, ILogger logger)
+        internal VsVersionProvider(IVsUIServiceOperation serviceOperation, ISetupConfigurationProvider setupConfigurationProvider, ILogger logger)
         {
-            Version = CalculateVersion(serviceProvider, setupConfigurationProvider, logger);
```

Comment:
I missed this type first time when I was compiling the list of non-free threaded constructors...